tests/test_annubes_env.py

```diff
@@ -100,7 +106,91 @@ def test_fix_time_types(time, expected_type):
+
+    This test checks:
+    1. If the observed probability of catch trials matches the specified probability
+    2. If the probability works correctly for various values including edge cases (0.0 and 1.0)
+    """
+    n_trials = 1000
+    env = AnnubesEnv(catch_prob=catch_prob, random_seed=RND_SEED)
+    catch_count = 0
+
+    for _ in range(n_trials):
+        env.reset()
+        trial_info = env.trial
+        if trial_info["catch"]:
+            catch_count += 1
+
+    observed_prob = catch_count / n_trials
+    expected_prob = catch_prob
+
+    # Check if the observed probability is close to the expected probability
+    assert np.isclose(
+        observed_prob,
+        expected_prob,
+        atol=0.05,
+    ), f"Expected catch probability {expected_prob}, but got {observed_prob}"
+
+
+@pytest.mark.parametrize(
+    ("session", "catch_prob", "max_sequential"),
+    [
+        ({"v": 0.5, "a": 0.5}, 0.5, 3),
+        ({"v": 0.3, "a": 0.7}, 0.1, 4),
+        ({"v": 0.3, "a": 0.3, "av": 0.3}, 0.7, 4),
+    ],
+)
+def test_annubes_env_max_sequential(session: dict, catch_prob: float, max_sequential: int) -> None:
+    """Test the maximum sequential trial constraint in the AnnubesEnv.
+
+    The test performs the following checks:
+    1. Ensures no sequence of non-catch trials longer than `max_sequential`
+       of the same type occurs.
+    2. Verifies that all specified trial types in the session occur.
+    3. Checks if the distribution of trial types is roughly balanced
+       according to the specified probabilities.
+    4. Verifies that the frequency of catch trials matches the specified
+       `catch_prob`.
+    """
+    env = AnnubesEnv(session=session, catch_prob=catch_prob, max_sequential=max_sequential, random_seed=RND_SEED)
+
+    trial_types: list[str | None] = []
+    for _ in range(N_TRIALS):
+        env.new_trial()
+        trial_types.append(env.trial["stim_type"])
+
+    # Check for sequences longer than max_sequential, excluding None (catch trials)
+    for i in range(len(trial_types) - max_sequential):
+        sequence = [t for t in trial_types[i : i + max_sequential + 1] if t is not None]
+        if len(sequence) > max_sequential:
+            assert len(set(sequence)) > 1, f"Found a sequence longer than {max_sequential} at index {i}"
+
+    # Check that all the trial types occur
+    assert set(trial_types) - {None} == set(env.session.keys()), "Not all trial types occurred"
+
+    # Check that the distribution is roughly balanced
+    expected_non_catch = N_TRIALS * (1 - catch_prob)
+
+    for trial_type, expected_prob in session.items():
+        count = trial_types.count(trial_type)
+        expected_count = expected_non_catch * expected_prob
+        # Allow for 20% deviation from expected count
+        assert (
+            0.8 * expected_count <= count <= 1.2 * expected_count
+        ), f"{trial_type} trials are not balanced. Expected: {expected_count:.2f}, Actual: {count}"
```

Comment:
this is an int (number expected).

```suggestion
        ), f"{trial_type} trials are not balanced. Expected: {expected_count}, Actual: {count}"
```